File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/CloudLandmarkDetector.java
package io.flutter.plugins.firebasemlvision;

import android.graphics.Rect;
import android.support.annotation.NonNull;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.ml.vision.FirebaseVision;
import com.google.firebase.ml.vision.cloud.FirebaseVisionCloudDetectorOptions;
import com.google.firebase.ml.vision.cloud.landmark.FirebaseVisionCloudLandmark;
import com.google.firebase.ml.vision.cloud.landmark.FirebaseVisionCloudLandmarkDetector;
import com.google.firebase.ml.vision.common.FirebaseVisionImage;
import com.google.firebase.ml.vision.common.FirebaseVisionLatLng;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import io.flutter.plugin.common.MethodChannel;

public class CloudLandmarkDetector implements Detector {
  public static final CloudLandmarkDetector instance = new CloudLandmarkDetector();

  private CloudLandmarkDetector() {}

  @Override
  public void handleDetection(
      FirebaseVisionImage image, Map<String, Object> options, final MethodChannel.Result result) {
    FirebaseVisionCloudDetectorOptions detectorOptions = FirebaseMlVisionPlugin.parseCloudDetectorOptions(options);
    FirebaseVisionCloudLandmarkDetector detector = FirebaseVision.getInstance().getVisionCloudLandmarkDetector(detectorOptions);

    detector
        .detectInImage(image)
        .addOnSuccessListener(
            new OnSuccessListener<List<FirebaseVisionCloudLandmark>>() {
              @Override
              public void onSuccess(List<FirebaseVisionCloudLandmark> firebaseVisionCloudLandmarks) {
                List<Map<String, Object>> landmarks =
                    new ArrayList<>(firebaseVisionCloudLandmarks.size());

                for (FirebaseVisionCloudLandmark landmark : firebaseVisionCloudLandmarks) {
                  Map<String, Object> landmarkData = new HashMap<>();
                  landmarkData.put("confidence", (double) landmark.getConfidence());
                  landmarkData.put("entityId", landmark.getEntityId());
                  landmarkData.put("landmark", landmark.getLandmark());

                  Rect boundingBox = landmark.getBoundingBox();
                  if (boundingBox != null) {
                    landmarkData.put("left", boundingBox.left);
                    landmarkData.put("top", boundingBox.top);
                    landmarkData.put("width", boundingBox.width());
                    landmarkData.put("height", boundingBox.height());
                  }

                  List<Map<String, Object>> latLngs = new ArrayList<>(landmark.getLocations().size());
                  for (FirebaseVisionLatLng latLng : landmark.getLocations()) {
                    Map<String, Object> latLngData = new HashMap<>(2);
                    latLngData.put("latitude", latLng.getLatitude());
                    latLngData.put("longitude", latLng.getLongitude());

                    latLngs.add(latLngData);
                  }

                  landmarkData.put("locations", latLngs);

                  landmarks.add(landmarkData);
                }

                result.success(landmarks);
              }
            }
        )
        .addOnFailureListener(
            new OnFailureListener() {
              @Override
              public void onFailure(@NonNull Exception e) {
                result.error("cloudLandmarkDetectorError", e.getLocalizedMessage(), null);
              }
            }
        );
  }
}

File: packages/firebase_ml_vision/ios/Classes/CloudLandmarkDetector.m
#import "FirebaseMlVisionPlugin.h"

@implementation LabelDetector
static FIRVisionCloudLandmarkDetector *detector;

+ (void)handleDetection:(FIRVisionImage *)image
                options:(NSDictionary *)options
                 result:(FlutterResult)result {
  FIRVision *vision = [FIRVision vision];
  FIRVisionCloudDetectorOptions *detectorOptions =
      [FLTFirebaseMlVisionPlugin parseCloudDetectorOptions:options];

  detector = [vision cloudLandmarkDetectorWithOptions:detectorOptions];

  [detector detectInImage:image
               completion:^(NSArray<FIRVisionCloudLandmark *> *_Nullable landmarks,
                            NSError *_Nullable error) {
                 if (error) {
                   [FLTFirebaseMlVisionPlugin handleError:error result:result];
                   return;
                 } else if (!landmarks) {
                   result(@[]);
                 }

                 NSMutableArray *landmarkData = [NSMutableArray array];
                 for (FIRVisionCloudLandmark *landmark in landmarks) {
                   __block NSMutableArray<NSDictionary *> *locations = [NSMutableArray array];
                   for (FIRVisionLatitudeLongitude *latLng in landmark.locations) {
                     [locations addObject:@{
                       @"latitude" : latLng.latitude,
                       @"longitude" : latLng.longitude,
                     }];
                   }

                   NSDictionary *data = @{
                     @"confidence" : landmark.confidence,
                     @"entityId" : landmark.entityId,
                     @"landmark" : landmark.landmark,
                     @"left" : @((int)landmark.frame.origin.x),
                     @"top" : @((int)landmark.frame.origin.y),
                     @"width" : @((int)landmark.frame.size.width),
                     @"height" : @((int)landmark.frame.size.height),
                     @"locations" : locations,
                   };

                   [landmarkData addObject:data];
                 }

                 result(landmarkData);
               }];
}
@end
